client/Components/BookATourForm.js

```diff
@@ -1,8 +1,139 @@
-import React from 'react';
+import React, { useState } from 'react';
+
+const BookATourForm = ( { handleFormChange } ) => {
+
+  const [isHidden, toggleHidden] = useState({
+    fullName: true,
+    email: true,
+    phone: true
+  });
+
+  const [numberOfPeople, changeNumberOfPeople] = useState(1);
+
+  const handleFormFieldFocus = (e) => {
+
+    const field = e.target.parentElement;
+    field.classList.add('focus');
+
+  }
+
+  const handleFieldBlur = (e, name) => {
+
+    const field = e.target.parentElement;
+    const newState = {...isHidden};
+    //can eventually add other helper functions here to do some *basic* front end validation
+    if (!e.target.value) {
+
+      if (field.classList.contains('complete')) field.classList.remove('complete');
+
+      field.classList.add('error');
+      newState[name] = false
+      toggleHidden(newState);
+
+    } else {
+
+      if (field.classList.contains('error')) field.classList.remove('error');
+
+      field.classList.add('complete');
+      newState[name] = true
+      toggleHidden(newState);
+
+    }
+
+  };
+
+  const handleMoveInClick = () => {
+
+    const cal = document.getElementById('calendar');
+    if (cal.hasAttribute('hidden')) {
+      cal.removeAttribute('hidden');
+    } else {
+      cal.setAttribute('hidden', true);
+    }
+  }
+
+  const handlePeopleChange = (e) => {
+    e.preventDefault();
+    let currentNumber = numberOfPeople;
+    const direction = e.target.classList[0];
+    const input = document.getElementById('desiredCapacity');
+    if (currentNumber === 1 && direction === 'down') return;
+    direction === 'up' ? currentNumber += 1 : currentNumber -=1;
+    input.value = currentNumber;
+    changeNumberOfPeople(currentNumber);
+  }
```

Comment:
handlePeopleChange might be better off being split into two seperate add/remove handlers.